flags/cxx17.cmake

```diff
@@ -0,0 +1,22 @@
+# Copyright (c) 2013, Ruslan Baratov
+# All rights reserved.
+
+if(DEFINED POLLY_FLAGS_CXX17_CMAKE)
+  return()
+else()
+  set(POLLY_FLAGS_CXX17_CMAKE 1)
+endif()
+
+include(polly_add_cache_flag)
+
+string(COMPARE EQUAL "${ANDROID_NDK_VERSION}" "" _not_android)
+
+
+# TODO: test other platfroms, CMAKE_CXX_FLAGS_INIT should work for all
+if(HUNTER_CMAKE_GENERATOR MATCHES "^Visual Studio.*$")
+  polly_add_cache_flag(CMAKE_CXX_FLAGS "/std:c++17")
```

Comment:
I think this trick is not working with Visual Studio generator. That's why all flags listed explicitly in `vs-mt` for example:
* https://github.com/ruslo/polly/blob/b9819d2e7f2be813b97654f6cbba20831751fa58/flags/vs-mt.cmake#L11-L14